test/e2e/cni/taint_controller_race.go

```diff
@@ -83,27 +84,29 @@ metadata:
 				Setup(cluster)
 			Expect(err).ToNot(HaveOccurred())
 
-			// assert pods demo-client and testserver are available on the node
-			clientPodName, err := PodNameOfApp(cluster, "demo-client", TestNamespace)
-			Expect(err).ToNot(HaveOccurred())
-
-			Eventually(func() (string, error) {
-				_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-					"curl", "-v", "-m", "3", "--fail", "test-server")
-				return stderr, err
-			}, "10s", "1s").Should(ContainSubstring("HTTP/1.1 200 OK"))
-
-			Eventually(func() (string, error) {
-				_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-					"curl", "-v", "-m", "3", "--fail", "test-server_kuma-test_svc_80.mesh")
-				return stderr, err
-			}, "10s", "1s").Should(ContainSubstring("HTTP/1.1 200 OK"))
-
-			Eventually(func() (string, error) { // should access a service with . instead of _
-				_, stderr, err := cluster.Exec(TestNamespace, clientPodName, "demo-client",
-					"curl", "-v", "-m", "3", "--fail", "test-server.kuma-test.svc.80.mesh")
-				return stderr, err
-			}, "10s", "1s").Should(ContainSubstring("HTTP/1.1 200 OK"))
+			Eventually(func(g Gomega) {
+				_, err := client.CollectResponse(
+					cluster, "demo-client", "test-server",
+					client.FromKubernetesPod(TestNamespace, "demo-client"),
+				)
+				g.Expect(err).ToNot(HaveOccurred())
```

Comment:
Maybe we could go with something like:
```suggestion
			Eventually(func(g Gomega) {
				g.Expect(client.CollectResponse(
					cluster, "demo-client", "test-server",
					client.FromKubernetesPod(TestNamespace, "demo-client"),
				)).Error().ToNot(HaveOccurred())
```
Wdyt?